crates/distribution-types/src/resolution.rs

```diff
@@ -117,3 +128,27 @@ impl From<Dist> for Requirement {
         }
     }
 }
+
+impl From<InstalledDist> for Requirement {
+    fn from(dist: InstalledDist) -> Self {
+        Self {
+            name: dist.name().clone(),
+            extras: vec![],
+            version_or_url: Some(pep508_rs::VersionOrUrl::VersionSpecifier(
+                pep440_rs::VersionSpecifiers::from(pep440_rs::VersionSpecifier::equals_version(
+                    dist.version().clone(),
+                )),
+            )),
+            marker: None,
+        }
+    }
+}
+
+impl From<ResolvedDist> for Requirement {
+    fn from(dist: ResolvedDist) -> Self {
+        match dist {
+            ResolvedDist::Installable(dist) => dist.clone().into(),
+            ResolvedDist::Installed(dist) => dist.clone().into(),
```

Comment:
Does this need `clone()`? It looks like it already takes an owned value?

Reply:
Yeah definitely not, thanks!